crates/bevy_platform/src/cfg.rs

```diff
@@ -0,0 +1,226 @@
+///
+/// ```ignore
+/// define_alias! {
+///     feature = "faster_algorithms" => { faster_algorithms }
+/// }
+/// ```
+///
+/// Now, `foo` can gate its usage of those faster algorithms on the alias, avoiding the need to
+/// expose its own "`faster_algorithms`" feature.
+/// This also avoids the unfortunate situation where one crate activates "`faster_algorithms`" on
+/// `bar` without activating that same feature on `foo`.
+///
+/// Once an alias is defined, there are 4 ways you can use it:
+///
+/// 1. Evaluate with no contents to return a `bool` indicating if the alias is active.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    if cfg::std!() {
+///        // Have `std`!
+///    } else {
+///        // No `std`...
+///    }
+///    ```
+/// 2. Pass a single code block which will only be compiled if the alias is active.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    cfg::std! {
+///        // Have `std`!
+///    # ()
+///    }
+///    ```
+/// 3. Pass a single `if { ... } else { ... }` expression to conditionally compile either the first
+///    or the second code block.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    cfg::std! {
+///        if {
+///            // Have `std`!
+///        } else {
+///            // No `std`...
+///        }
+///    }
+///    ```
+/// 4. Use in a [`switch`] arm for more complex conditional compilation.
+///    ```rust
+///    # use bevy_platform::cfg;
+///    cfg::switch! {
+///        cfg::std => {
+///            // Have `std`!
+///        }
+///        cfg::alloc => {
+///            // No `std`, but do have `alloc`!
+///        }
+///        _ => {
+///            // No `std` or `alloc`...
+///        }
+///    }
+///    ```
+#[doc(inline)]
+pub use crate::define_alias;
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! switch {
+    ({ $($tt:tt)* }) => {{
+        $crate::switch! { $($tt)* }
+    }};
+    (_ => { $($output:tt)* }) => {
+        $($output)*
+    };
+    (
+        $cond:path => $output:tt
+        $($( $rest:tt )+)?
+    ) => {
+        $cond! {
+            if {
+                $crate::switch! { _ => $output }
+            } else {
+                $(
+                    $crate::switch! { $($rest)+ }
+                )?
+            }
+        }
+    };
+    (
+        #[cfg($cfg:meta)] => $output:tt
+        $($( $rest:tt )+)?
+    ) => {
+        #[cfg($cfg)]
+        $crate::switch! { _ => $output }
+        $(
+            #[cfg(not($cfg))]
+            $crate::switch! { $($rest)+ }
+        )?
+    };
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! noop {
+    () => { false };
+    (if { $($p:tt)* } else { $($n:tt)* }) => { $($n)* };
+    ($($p:tt)*) => {};
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! pass {
+    () => { true };
+    (if { $($p:tt)* } else { $($n:tt)* }) => { $($p)* };
+    ($($p:tt)*) => { $($p)* };
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! define_alias {
+    ($(
+        $meta:meta => {
```

Comment:
```suggestion
        #[cfg($meta:meta)] => {
```

I would much prefer this be `#[cfg(...)]`, as it signals what this argument is translating to. Before, it may be confused for a variable assignment, but using an explicit `#[cfg(...)]` tells users that they can use things like `all()` and `any()` here. (This is also more consistent with `cfg::switch!`!)

```rust
// before
define_alias! {
    feature = "alloc" => {
        /// Indicates the `alloc` crate is available and can be used.
        alloc
    }
}

// After
define_alias! {
    #[cfg(feature = "alloc")] => {
        /// Indicates the `alloc` crate is available and can be used.
        alloc
    }
}
```

This syntax highlighting is also better :)

Reply:
Yeah I agree. My first thought was less syntax would be nicer, but familiarity is better in this case, since we don't expect users to actually call this macro more than once per crate.